k8s-operator/apis/v1alpha1/types_tsdnsconfig.go

```diff
@@ -19,12 +19,43 @@ var DNSConfigKind = "DNSConfig"
 // +kubebuilder:resource:scope=Cluster,shortName=dc
 // +kubebuilder:printcolumn:name="NameserverIP",type="string",JSONPath=`.status.nameserver.ip`,description="Service IP address of the nameserver"
 
+// DNSConfig can be deployed to cluster to make a subset of Tailscale MagicDNS
+// names resolvable by cluster workloads. Use this if: A) you need to refer to
+// tailnet services, exposed to cluster via Tailscale Kubernetes operator egress
+// proxies by the MagicDNS names of those tailnet services (usually because the
+// services run over HTTPS)
+// B) you have exposed a cluster workload to the tailnet using Tailscale Ingress
+// and you also want to refer to the workload from within the cluster over the
+// Ingress's MagicDNS name (usually because you have some callback component
+// that needs to use the same URL as that used by a non-cluster client on
+// tailnet).
+// When a DNSConfig is applied to a cluster, Tailscale Kubernetes operator will
+// deploy a nameserver for ts.net DNS names and automatically populate it with records
+// for any Tailscale egress or Ingress proxies deployed to that cluster.
+// Currently you must manually update your cluster DNS configuration to add the
+// IP address of the deployed nameserver as a ts.net stub nameserver.
+// Instructions for how to do it:
+// https://kubernetes.io/docs/tasks/administer-cluster/dns-custom-nameservers/#configuration-of-stub-domain-and-upstream-nameserver-using-coredns (for CoreDNS),
+// https://cloud.google.com/kubernetes-engine/docs/how-to/kube-dns (for kube-dns).
+// Tailscale Kubernetes operator will write the address of a Service fronting
+// the nameserver to dsnconfig.status.nameserver.ip.
+// DNSConfig is a singleton - you must not create more than one.
+// NB: if you want cluster workloads to be able to refer to Tailscale Ingress
+// using its MagicDNS name, you must also annotate the Ingress resource with
+// tailscale.com/experimental-forward-cluster-traffic-via-ingress annotation to
+// ensure that the proxy created for the Ingress listens on its Pod IP address.
+// NB: IPv6 only clusters are currently not supported.
```

Comment:
As a first time reader, I'm not sure if this is saying that IPv6 is not supported at all, or if it's just not supported to remove IPv4 addresses from the name server.

Reply:
Thanks, I've reworded it - it's meant to state that clusters with IPv6 Pod CIDR ranges are not supported